src/agent/agent_info/src/agent_info_persistance.cpp

```diff
@@ -157,11 +205,37 @@ void AgentInfoPersistance::SetUUID(const std::string& uuid)
     SetAgentInfoValue("uuid", uuid);
 }
 
+void AgentInfoPersistance::SetGroups(const std::vector<std::string>& groupList)
+{
+    try
+    {
+        SQLite::Transaction transaction(*m_db);
+
+        m_db->exec("DELETE FROM " + AGENT_GROUP_TABLE_NAME + ";");
```

Comment:
We should test this part too, for instance, set the groups to 1,2, verify, and then set to 2,3 and verify that 1 is no longer there.

Reply:
Done!